[BUG] Unable to add any ticker

Hey,
I installed latest version 3.7.2. When calling basic ticker function, I get an error:
root@name1:~/discord-stock-ticker# ./discord-stock-ticker
INFO[0000] Running discord-stock-ticker version v3.7.2...
WARN[0000] Will not be storing state.
INFO[0000] Starting api server on 0.0.0.0:8080...
INFO[0002] Added crypto: ethereum
panic: non-positive interval for NewTicker

goroutine 21 [running]:
time.NewTicker(0x0, 0x64)
        /usr/local/go/src/time/tick.go:24 +0x151
main.(*Ticker).watchCryptoPrice(0xc0001ae2d0)
        /home/riley/go/src/github.com/rssnyder/discord-stock-ticker/ticker.go:487 +0xb51
created by main.(*Ticker).Start
        /home/riley/go/src/github.com/rssnyder/discord-stock-ticker/ticker.go:95 +0x3f


This is basic ticker function I called:
curl -X POST -H "Content-Type: application/json" --data '{
  "name": "ethereum",
  "crypto": true,
  "set_nickname": true,
  "discord_bot_token": "my bot token"
}' localhost:8080/ticker

I disabled FW to make sure its not connection issue, same issue.

--- a/board_request.go
+++ b/board_request.go
@@ -96,6 +96,11 @@ func (m *Manager) AddBoard(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure frequency is set
+	if boardReq.Frequency <= 0 {
+		boardReq.Frequency = 60
+	}
+
 	// ensure name is set
 	if boardReq.Name == "" {
 		logger.Error("Board Name required")

--- a/gas.go
+++ b/gas.go
@@ -72,8 +72,9 @@ func (g *Gas) watchGasPrice() {
 	}
 
 	// check for frequency override
+	// set to one hour to avoid lockout
 	if *frequency != 0 {
-		g.Frequency = *frequency
+		g.Frequency = 600
 	}
 
 	ticker := time.NewTicker(time.Duration(g.Frequency) * time.Second)

--- a/gas_request.go
+++ b/gas_request.go
@@ -78,6 +78,11 @@ func (m *Manager) AddGas(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure frequency is set
+	if gasReq.Frequency <= 0 {
+		gasReq.Frequency = 60
+	}
+
 	// ensure network is set
 	if gasReq.Network == "" {
 		logger.Error("Network required")

--- a/holders_request.go
+++ b/holders_request.go
@@ -80,6 +80,11 @@ func (m *Manager) AddHolders(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure frequency is set
+	if holdersReq.Frequency <= 0 {
+		holdersReq.Frequency = 60
+	}
+
 	// ensure network is set
 	if holdersReq.Network == "" {
 		logger.Error("Network required")

--- a/marketcap_request.go
+++ b/marketcap_request.go
@@ -91,6 +91,11 @@ func (m *Manager) AddMarketCap(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure frequency is set
+	if stockReq.Frequency <= 0 {
+		stockReq.Frequency = 60
+	}
+
 	// ensure currency is set
 	if stockReq.Currency == "" {
 		stockReq.Currency = "usd"

--- a/ticker.go
+++ b/ticker.go
@@ -552,6 +552,10 @@ func (s *Ticker) watchCryptoPrice() {
 				fmtPrice = fmt.Sprintf("%s%.10f", s.CurrencySymbol, priceData.MarketData.CurrentPrice.USD)
 			case 11:
 				fmtPrice = fmt.Sprintf("%s%.11f", s.CurrencySymbol, priceData.MarketData.CurrentPrice.USD)
+			case 12:
+				fmtPrice = fmt.Sprintf("%s%.12f", s.CurrencySymbol, priceData.MarketData.CurrentPrice.USD)
+			case 13:
+				fmtPrice = fmt.Sprintf("%s%.13f", s.CurrencySymbol, priceData.MarketData.CurrentPrice.USD)
 			default:
 
 				// Check for cryptos below 1c

--- a/ticker_request.go
+++ b/ticker_request.go
@@ -97,6 +97,11 @@ func (m *Manager) AddTicker(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure frequency is set
+	if stockReq.Frequency <= 0 {
+		stockReq.Frequency = 60
+	}
+
 	// ensure currency is set
 	if stockReq.Currency == "" {
 		stockReq.Currency = "usd"

--- a/token_request.go
+++ b/token_request.go
@@ -90,6 +90,11 @@ func (m *Manager) AddToken(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// ensure frequency is set
+	if tokenReq.Frequency <= 0 {
+		tokenReq.Frequency = 60
+	}
+
 	// ensure network is set, default to eth
 	if tokenReq.Network == "" {
 		tokenReq.Network = "ethereum"

--- a/utils/ethgaswatch.go
+++ b/utils/ethgaswatch.go
@@ -1,0 +1,74 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+const (
+	EthGasWatchURL = "https://ethgas.watch/api/gas"
+)
+
+type EthGasWatchData struct {
+	Slow struct {
+		Gwei int     `json:"gwei"`
+		Usd  float64 `json:"usd"`
+	} `json:"slow"`
+	Normal struct {
+		Gwei int     `json:"gwei"`
+		Usd  float64 `json:"usd"`
+	} `json:"normal"`
+	Fast struct {
+		Gwei int     `json:"gwei"`
+		Usd  float64 `json:"usd"`
+	} `json:"fast"`
+	Instant struct {
+		Gwei int     `json:"gwei"`
+		Usd  float64 `json:"usd"`
+	} `json:"instant"`
+	Ethprice    float64 `json:"ethPrice"`
+	Lastupdated int64   `json:"lastUpdated"`
+	Sources     []struct {
+		Name       string `json:"name"`
+		Source     string `json:"source"`
+		Fast       int    `json:"fast"`
+		Standard   int    `json:"standard"`
+		Slow       int    `json:"slow"`
+		Lastblock  int    `json:"lastBlock,omitempty"`
+		Instant    int    `json:"instant,omitempty"`
+		Lastupdate int64  `json:"lastUpdate,omitempty"`
+	} `json:"sources"`
+}
+
+func GetEthGasWatchData() (EthGasWatchData, error) {
+
+	var prices EthGasWatchData
+
+	req, err := http.NewRequest("GET", EthGasWatchURL, nil)
+	if err != nil {
+		return prices, err
+	}
+
+	req.Header.Add("User-Agent", "Mozilla/5.0")
+	req.Header.Add("accept", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return prices, err
+	}
+
+	results, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return prices, err
+	}
+
+	err = json.Unmarshal(results, &prices)
+	if err != nil {
+		fmt.Printf(resp.Status)
+		return prices, err
+	}
+
+	return prices, nil
+}

--- a/utils/gas.go
+++ b/utils/gas.go
@@ -1,0 +1,24 @@
+package utils
+
+type GasData struct {
+	Standard int `json:"standard"`
+	Fast     int `json:"fast"`
+	Instant  int `json:"instant"`
+}
+
+// get gas prices based on network
+func GetGasPrices(network string) (GasData, error) {
+	switch network {
+	case "polygon":
+		return GetZapperData(network)
+	case "binance-smart-chain":
+		return GetZapperData(network)
+	default:
+		result, err := GetEthGasWatchData()
+		return GasData{
+			Standard: result.Normal.Gwei,
+			Fast:     result.Fast.Gwei,
+			Instant:  result.Instant.Gwei,
+		}, err
+	}
+}

--- a/utils/zapper.go
+++ b/utils/zapper.go
@@ -8,21 +8,15 @@ import (
 )
 
 const (
-	GasURL = "http://api.zapper.fi/v1/gas-price?network=%s&api_key=%s"
-	apiKey = "96e0cc51-a62e-42ca-acee-910ea7d2a241"
+	ZapperURL = "http://api.zapper.fi/v1/gas-price?network=%s&api_key=%s"
+	apiKey    = "96e0cc51-a62e-42ca-acee-910ea7d2a241"
 )
 
-type GasPrices struct {
-	Standard int `json:"standard"`
-	Fast     int `json:"fast"`
-	Instant  int `json:"instant"`
-}
+func GetZapperData(network string) (GasData, error) {
 
-func GetGasPrices(network string) (GasPrices, error) {
+	var prices GasData
 
-	var prices GasPrices
-
-	reqUrl := fmt.Sprintf(GasURL, network, apiKey)
+	reqUrl := fmt.Sprintf(ZapperURL, network, apiKey)
 
 	req, err := http.NewRequest("GET", reqUrl, nil)
 	if err != nil {

